### DISCUSSION TITLE
How do I use a custom position for the gravity command?

### DISCUSSION THREAD

**Discussion by timesword3d** [ImageMagick/ImageMagick#3243] [Help]
I'm not sure if this is really possible, or if I'm going about it the wrong way, but looking around online, it seems like you should be able to use custom positions for gravity commands by using it in conjunction with the **geometry** command. When I do things like: ``convert 1.png -background none -gravity southwest -extent 40x40 2.png``, it properly expands the canvas on the north and east side of the image. However, if I add ``-geometry +10+0`` after the gravity command for example, I get a very messed up image. Ideally, I want to be able to offset the gravity by a **percentage** of the total pixels, but I can't even figure out how to tell it to offset a certain amount of pixels with geometry. I don't want **extent** to equally distribute pixels on each side, I want to pick a specific point of gravity so that it aims to make that point the center.

Would really appreciate any help. My discord is **Phil#6666** if you would rather DM me.

**Comment by snibgo** [comment]
What version of IM, on what platform?

If you use IM v6, I suggest you upgrade to v7. Then, use "magick" instead of "convert".

Then you can use expressions instead of numbers, eg in "-geometry" settings and "-extent" operations. For example:
```
magick toes.png -gravity southwest -extent %[fx:w*2]x%[fx:h*2]-%[fx:w*0.1]-100 x.png
```
This doubles the width and height, with a gap at the left of the original width times 0.1, and a gap at the bottom of 100 pixels.

**Comment by LeandroFigueredo** [comment]
Hello, Thanks for this discussion, I'm experiencing the same problem.

@timesword3d Has there been any progress on this issue?

**Reply by fmw42** [reply]
What is your exact issue -- can you provide a description of your desired processing?  What did you try? What is your IM version and platform? Have you tried snibgo's solution?

**Reply by LeandroFigueredo** [reply]
Thanks for the answer,

I'm using IM v6 and, in my case, I need to compose a lot of images into a final image with a transparent background. I have all the geometries and gravities, but when composing one by one the geometry is applied but the gravity is ignored. I'm using a Ruby MiniMagick library, but first I try to reproduce the result in IM directly and then write the code in ruby. I'm running this on Ubuntu 18.04. And in my case, I can't change the IM version to 7 because it's legacy code that I can't change. Sorry for the lack of information before. To resume, I need to combine gravity with geometry in any composition. What is the best way to do this?

**Reply by fmw42** [reply]
Can you provide an example of your IM command line code?  Review the following link and see if that helps.

https://legacy.imagemagick.org/Usage/layers/#convert
